services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/dto/Role.java

```diff
@@ -0,0 +1,87 @@
+package com.sourcefuse.jarc.services.usertenantservice.dto;
+
+import com.sourcefuse.jarc.services.usertenantservice.commons.UserModifiableEntity;
+import com.sourcefuse.jarc.services.usertenantservice.enums.RoleType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.UUID;
+
+@Getter
+@Setter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "roles", schema = "main")
+public class Role extends UserModifiableEntity implements Serializable {
```

Comment:
Why Serializable

Reply:
elow are the reason but if our requirement is different we can remove it ,please suggest .

Data Exchange: DTOs are commonly used to transfer data between different parts of an application or across network boundaries
Caching and Persistence: In some scenarios, DTOs may need to be cached or persisted for later retrieval.
some disadvantages:
Limited Data Exchange: Without serialization, DTOs cannot be easily transmitted or shared between different systems or components.
Without serialization, you would need to find alternative mechanisms to transfer the data, such as manually converting the DTOs into a different format (e.g., JSON, XML) and handling the data transfer manually.